client/state/sites/reducer.js

```diff
@@ -103,8 +103,6 @@ export function items( state = {}, action ) {
 			}, initialNextState );
```

Comment:
@aduth of course. This is really an edge case though, let me clarify what happens and why we need to keep that update behavior:

* We load `/jetpack/connect`.
* The `<QuerySites />` in layout.jsx issues a request to `/me/sites`.
* We connect a site, and add it to the sites state so we can use it.
* The `/me/sites` request is old, and does not contain that site yet; so when that request finishes, that site is removed from the state.

In that case we want to keep that newly connected site instead of removing it.